src/components/index.cjsx

```diff
@@ -38,13 +40,37 @@ Dashboard = React.createClass
 SingleTask = React.createClass
   mixins: [Router.State, LoadableMixin]
 
+  getInitialState: ->
+    {courseId, id} = @getParams()
+    # not the best way to determine is practice...wanted to use route name but that also seems risky
+    isPractice = not id?
+
+    state =
+      isPractice: isPractice
+      taskId: if isPractice then CourseStore.getPracticeId(courseId) else id
+      id: if isPractice then courseId else id
+      type: if isPractice then 'practice' else 'task'
+
+  update: ->
+    @setState({taskId: CourseStore.getPracticeId(@getParams().courseId)}) unless @state.taskId
+
   getFlux: ->
-    store: TaskStore
-    actions: TaskActions
+
+    fluxes =
+      task:
+        store: TaskStore
+        actions: TaskActions
+      practice:
+        store: CourseStore
+        actions: CourseActions
+
+    fluxes[@state.type]
```

Comment:
Hmm, this looks interesting. Would there be less conditional logic in here if the component was split into 2? (maybe in a subsequent PR since I'm currently waiting on the CourseStore to land?)